README.md

```diff
@@ -96,7 +96,17 @@ FAR CRs are deleted by NHC after it sees the Node is healthy again.
 
 #### Example CR
 
-The FAR, `FenceAgentsRemediation`, CR is created by the admin and is used to trigger the fence-agent on a specific node. The CR includes an *agent* field for the fence-agent name, *sharedparameters* field with all the shared, not specific to a node, parameters, and *nodeparameters* field to specify the parameters for the fenced node.
+The FAR CR (CustomResource), created by the admin, is used to trigger the fence-agent on a specific node.
+
+The CR includes the following parameters:
+
+* `agent` - the fence-agent name
+* `sharedparameters` - common cluster nodes information for executing the fence agent
+* `nodeparameters` - specific cluster nodes information for executing the fence agent
```

Comment:
Mentioning the next three parameters as optional and not mentioning these two as optional means that they are not optional.
In practice they are [optional](https://github.com/medik8s/fence-agents-remediation/blob/main/api/v1alpha1/fenceagentsremediation_types.go#L80-L86), even though I don't see a case where a user won't use/set them, thus I am leaning towards setting them as non-optional parameters or simply remove the `omitempty`. WDYT?

Reply:
I guess they are optional just because it's up to the user to decide which specific parameters they want, but you're right that this way is confusing. I might rephrase the last three as "If unset, the default value is..." ?

Reply:
> it's up to the user to decide which specific parameters they want

and IF they want it. I think it is ok to keep them optional

Reply:
>"If unset, the default value is..." ?

Or just remove the _optional_ words. `Optional parameter, default is "5s".` -> `Default is "5s".`
I don't have a strong preference here.

> I think it is ok to keep them optional

Agreed. My question was more about the `sharedparameters` and `nodeparameters` which maybe should not be optional. Without them running the fence agent would fail as it misses some parameters.

Reply:
I was referring to sharedparameters and nodeparameters too :)

> Without them running the fence agent would fail as it misses some parameters

but I didn't realized that. They're surely not optional then

Reply:
>They're surely not optional then

Yes, I was just looking for another point of view to assert my assumption for taking an action :)